crates/engine/tree/src/tree/payload_processor/prewarm.rs

```diff
@@ -123,6 +132,15 @@ where
         self.ctx.metrics.cache_saving_duration.set(start.elapsed().as_secs_f64());
     }
 
+    /// Removes the `actions_tx` currently stored in the struct, replacing it with a new one that
+    /// does not point to any active receiver.
+    ///
+    /// This is used to drop the `actions_tx` after all tasks have been spawned, and should not be
+    /// used in any context other than the `run` method.
+    fn drop_actions_tx(&mut self) {
+        self.actions_tx = channel().0;
```

Comment:
I think this is fine, less invasive than using an option which would then require a bunch of .expect calls